### PR TITLE
Fix Line2D.TryIntersect to pass nullable Point2D

### PR DESCRIPTION
**Pull request by f-frhs**
This pull-request is asking for help to solve #178.

The problem is that `LineSegment2D.TryIntersect()` always passes by reference `Point2D` of intersection, regardless of whether the two `LineSegment2D` actually have a valid intersection or not.

In my fix, `LineSegment2D.TryIntersect()` passes by ref `Point2D?`, which can indicate that there is no intersection.
My fix used #111 as reference.

I'm not sure that this breaking change is acceptable or not.
If there is another way to solve this, I'm happy to learn it.

I considered these fix-plans:
1. Leave it as it is
2. Pass by ref default(Point2D)
3. Pass by ref Nullable&lt;Point2D&gt;
4. Pass by ref Point2D.Null (kinda)
5. Return Nullable&lt;Point2D&gt; instead of bool

### CODE REVIEW

File: src/Spatial/Euclidean/LineSegment2D.cs [hunk 1]
```diff
@@ -134,19 +134,19 @@ public Point2D ClosestPointTo(Point2D p)
...
-        /// <param name="intersection">When this method returns, contains the intersection point, if the conversion succeeded, or the default point if the conversion failed.</param>
-        /// <param name="tolerance">The tolerance used when checking if the lines are parallel</param>
+        /// <param name="other">The other line segment to compute the intersection with</param>
+        /// <param name="intersection">The intersection if it exists; otherwise null</param>
+        /// <param name="tolerance">The tolerance used when checking if the line segments are parallel</param>
         /// <returns>True if an intersection exists; otherwise false</returns>
         [Pure]
-        public bool TryIntersect(LineSegment2D other, out Point2D intersection, Angle tolerance)
+        public bool TryIntersect(LineSegment2D other, out Point2D? intersection, Angle tolerance)
         {
```
Comment (jkalias) [inline src/Spatial/Euclidean/LineSegment2D.cs:146]: Initialization ```intersection = null;```

File: src/Spatial/Euclidean/LineSegment2D.cs [hunk 2]
```diff
@@ -134,19 +134,19 @@ public Point2D ClosestPointTo(Point2D p)
...
             if (this.IsParallelTo(other, tolerance))
             {
-                intersection = default(Point2D);
+                intersection = null;
```
Comment (jkalias) [inline src/Spatial/Euclidean/LineSegment2D.cs:149]: Not needed here if the initialization is done at the function entry point

File: src/Spatial/Euclidean/LineSegment2D.cs [hunk 3]
```diff
@@ -159,8 +159,17 @@ public bool TryIntersect(LineSegment2D other, out Point2D intersection, Angle to
...
 
-            intersection = p + (t * r);
-            return (0.0 <= t && t <= 1.0) && (0.0 <= u && u <= 1.0);
+            var isIntersected = (0.0 <= t && t <= 1.0) && (0.0 <= u && u <= 1.0);
+            if (isIntersected)
+            {
+                intersection = p + (t * r);
+                return true;
+            }
+            else
```
Comment (jkalias) [inline src/Spatial/Euclidean/LineSegment2D.cs:168]: You can simplify all this by simply ```return intersection != null;``` and completely removing the ```else``` case.

**Review by jkalias — CHANGES_REQUESTED** [review]

**Review by jkalias — APPROVED** [review]

File: src/Spatial/Euclidean/LineSegment2D.cs (hunk 3 above)
Comment (f-frhs) [reply src/Spatial/Euclidean/LineSegment2D.cs]: Thanks for a comment.

How about the following case?
83840266-f4d1c880-a6b2-11ea-8137-56f9fee366af

This comes from #178.

Comment (f-frhs) [reply src/Spatial/Euclidean/LineSegment2D.cs]: Thanks for a comment.
Your understanding is correct that we can verify it by using `LineSegment2DTests.IntersectWithTest2`.

I'm happy with the merged code. Thanks.

Comment (jkalias) [reply src/Spatial/Euclidean/LineSegment2D.cs]: Thank YOU for opening the issue and providing a PR 😃

**Pull request merged**